Studies/week_7/ex-7-3-1/crtp.h

```diff
@@ -0,0 +1,79 @@
+#if !defined(CRTP)
+#define CRTP
+
+#include <iostream>
+#include <ostream>
+#include <typeinfo>
+
+
+// Base class
+class Animal
+{
+public:
+    std::string noise {"animal"};
+    Animal() {}
+    ~Animal() {}
+    virtual void make_noise(){std::cout << noise << "\n";};
```

Comment:
OK when testing stuff (which I assume was the case here), but it rarely makes sense to combine virtual with CRTP.